java/client/src/org/openqa/selenium/remote/internal/OkHttpClient.java

```diff
@@ -116,11 +117,20 @@ public void close() {
 
     @Override
     public HttpClient createClient(URL url) {
-      okhttp3.OkHttpClient client = new okhttp3.OkHttpClient.Builder()
+      okhttp3.OkHttpClient.Builder builder = new okhttp3.OkHttpClient.Builder()
           .connectionPool(pool)
           .followRedirects(true)
-          .followSslRedirects(true)
-          .build();
+          .followSslRedirects(true);
+
+      String urlUserInfo = url.getUserInfo();
+      if (urlUserInfo != null) {
+        UserInfo userInfo = new UserInfo(urlUserInfo);
```

Comment:
I'm not sure we need this new class. It's enough to just split the string and grab the bits.

Reply:
there is exactly the same logic (and exactly the same issue) in Apache HTTP client, so IMHO it worth following DRY principle